algebra/src/fft/polynomial/dense.rs

```diff
@@ -413,9 +426,12 @@ impl<'a, 'b, F: PrimeField> Mul<&'a DensePolynomial<F>> for &'b DensePolynomial<
         if self.is_zero() || other.is_zero() {
             DensePolynomial::zero()
         } else {
-            let mut res = self.clone();
-            <DensePolynomial<F> as MulAssign<&DensePolynomial<F>>>::mul_assign(&mut res, other);
-            res
+            let domain = get_best_evaluation_domain(self.coeffs.len() + other.coeffs.len())
+                .expect("Field is not smooth enough to construct domain");
+            let mut self_evals = self.evaluate_over_domain_by_ref(domain.clone());
+            let other_evals = other.evaluate_over_domain_by_ref(domain);
+            self_evals *= &other_evals;
+            self_evals.interpolate()
```

Comment:
What do you mean ? It is common, when implementing arithmetic operators, to provide operands params in different combinations (by reference and/or by value), so to avoid explicit clones. You'll find some dereferencing that, can be faster or, in the worst case, as much performant as clones